src/Place/Canonical/CanonicalService.php

```diff
@@ -0,0 +1,88 @@
+<?php
+
+declare(strict_types=1);
+
+namespace CultuurNet\UDB3\Place\Canonical;
+
+use CultuurNet\UDB3\Event\ReadModel\Relations\RepositoryInterface;
+use CultuurNet\UDB3\Label\ReadModels\Relations\Repository\ReadRepositoryInterface;
+use CultuurNet\UDB3\Label\ValueObjects\LabelName;
+use CultuurNet\UDB3\Label\ValueObjects\RelationType;
+use CultuurNet\UDB3\ReadModel\DocumentRepository;
+
+class CanonicalService
+{
+    private string $museumpasLabel;
+
+    private RepositoryInterface $eventRelationsRepository;
+
+    private ReadRepositoryInterface $labelRelationsRepository;
+
+    private DocumentRepository $placeRepository;
+
+    public function __construct(
+        string $museumpasLabel,
+        RepositoryInterface $eventRelationsRepository,
+        ReadRepositoryInterface $labelRelationsRepository,
+        DocumentRepository $placeRepository
+    ) {
+        $this->museumpasLabel = $museumpasLabel;
+        $this->eventRelationsRepository = $eventRelationsRepository;
+        $this->labelRelationsRepository = $labelRelationsRepository;
+        $this->placeRepository = $placeRepository;
+    }
+
+    public function getCanonical(array $placeIds): string
+    {
+        $placesWithMuseumpas = $this->getPlacesWithMuseumPasInCluster($placeIds);
+        if (count($placesWithMuseumpas) === 1) {
```

Comment:
I think we can throw a custom exception if `placesWithMuseumpas` is bigger than 1. The other code can stay.